pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -137,12 +137,8 @@ func (w Workflows) Filter(predicate WorkflowPredicate) Workflows {
 
 // GetTTLStrategy return TTLStrategy based on Order of precedence:
 //1. Workflow, 2. WorkflowTemplate, 3. Workflowdefault
-func (w *Workflow) GetTTLStrategy(defaultTTLStrategy *TTLStrategy) *TTLStrategy {
+func (w *Workflow) GetTTLStrategy() *TTLStrategy {
```

Comment:
As a new approach, `StoredWorkflowSpec` will hold the merged workflowSpec which will have WFSpec, WFTSpec, and WFDefaultSpec. Reverting the previous fix